EpiAware/src/EpiObsModels/ObservationErrorModels/AbstractTuringObservationModel.jl

```diff
@@ -0,0 +1,29 @@
+abstract type AbstractTuringObservationErrorModel <: AbstractTuringObservationModel end
+
+@model function EpiAwareBase.generate_observations(
+        obs_model::AbstractTuringObservationErrorModel,
+        y_t,
+        Y_t)
+    @submodel priors = generate_observation_error_priors(obs_model, y_t, Y_t)
+
+    if ismissing(y_t)
+        y_t = Vector{Union{Real, Missing}}(missing, length(Y_t))
+    end
+
+    first_Y_t = findfirst(!ismissing, Y_t)
+    for i in first_Y_t:length(Y_t)
```

Comment:
We could now also fairly easily add support for using a vectorised observation model in cases where we know there are no missing elements (or all are missing). I don't think we know if this has performance implications though.